pkg/master/controller.go

```diff
@@ -265,50 +349,48 @@ func (c *Controller) ReconcileEndpoints(serviceName string, ip net.IP, endpointP
 		}
 	}
 
-	// First, determine if the endpoint is in the format we expect (one
-	// subset, ports matching endpointPorts, N IP addresses).
-	formatCorrect, ipCorrect, portsCorrect := checkEndpointSubsetFormat(e, ip.String(), endpointPorts, c.MasterCount, reconcilePorts)
+	// ... and the list of master IP keys from etcd
+	masterIPs, err := c.MasterLeases.ListLeases()
+	if err != nil {
+		return err
+	}
+
+	// assume that no endpoint in storage is an invalid state (we just listed
+	// our own IP), and thus we should not update based on the results
```

Comment:
NIT: if you do another pass, this comment could use a bit of rephrasing (and yes, IIRC this is one of the ones that I wrote):  

> Since we just refreshed our own key, assume that zero endpoints returned from storage indicates an issue or invalid state, and thus do not update the endpoints list based on the result.